tools/runqslower.py

```diff
@@ -161,25 +161,23 @@
 RAW_TRACEPOINT_PROBE(sched_switch)
 {
     // TP_PROTO(bool preempt, struct task_struct *prev, struct task_struct *next)
-    struct task_struct *prev = (struct task_struct *)ctx->args[1];
-    struct task_struct *next= (struct task_struct *)ctx->args[2];
-    u32 pid, tgid;
+    u32 pid;
     long state;
+    struct task_struct *next = (struct task_struct *)ctx->args[2];
+    struct task_struct *prev = (struct task_struct *)ctx->args[1];
+
+    bpf_probe_read(&pid, sizeof(next->pid), &next->pid);
```

Comment:
This does not sound right. The pid could be overwritten by later at line 174. I suggest keeping code as is, just removing the dead code at line 172 to bpf_probe_read value for tgid.

Reply:
Understood, you are correct about line 174 overwriting the pid. I will fix that. Might as well move the line 169 after the block:
```
    if (state == TASK_RUNNING) {
        bpf_probe_read(&pid, sizeof(prev->pid), &prev->pid);
        if (!(FILTER_PID || pid == 0)) {
            u64 ts = bpf_ktime_get_ns();
            start.update(&pid, &ts);
        }
    }

```
> just removing the dead code at line 172 to bpf_probe_read value for tgid.
Line 172:
```
bpf_probe_read(&state, sizeof(long), (const void *)&prev->state);
```
The above does not seem dead code to me. If I remove line 172, the `state` variable will stay undefined. I might missing something here.

Also the `tgid` is not used at all. So why introduce is back?

Reply:
I mean tgid code. The 'state' bpf_probe_read certainly useful. The line number sometimes refers to old, sometimes refer to new code, so maybe confusing.

Reply:
I think just remove dead code and use bpf_probe_read_str() change should be good enough.